### PR TITLE
Selected server google account

### LINKED ISSUE #1786: Choose an account dialog starts with no option initially selected.
#### Software and hardware versions 
Collect v1.12.x, Android v5.1.1, device used Oppo f1

#### Problem description
The "Choose an account" dialog starts without any option initially selected.

#### Steps to reproduce the problem
General settings -> Server -> Set type to Google drive,sheets -> Now select Google account

#### Expected behavior
The dialog should start with the previously selected google account (if it exists) or the first option.

#### Other information 
Found while working on #1777 
Certainly not a problem but most app have one option as a default one in radio options dialog.

### PR DESCRIPTION
**Pull request by ghost**
Closes #1786 

#### What has been done to verify that this works as intended?

I tested manually the possible situations opening the Google Account Picker Dialog and all of them worked properly:

- Opening the dialog without any Google account added redirects to the Google login page.
- Opening the dialog with one or more Google accounts added but none of them selected checks the first one.
- Opening the dialog with one or more Google accounts added and one of them selected checks the one that was selected.

#### Why is this the best possible solution? Were any other approaches considered?

This is the only approach I considered, the intent that starts the Picker Dialog  just needed to be updated before starting the activity.

#### Are there any risks to merging this code? If so, what are they?

I wrote just a few lines of code that check if there is a Google account stored in Preferences, so I don't think there is any risk merging the code.

#### Do we need any specific form for testing your changes? If so, please attach one.

It can be tested with a device/emulator following the 'Steps to reproduce the problem' of the issue.

### CODE REVIEW

File: collect_app/src/main/java/org/odk/collect/android/utilities/gdrive/GoogleAccountsManager.java [hunk 1]
```diff
@@ -171,6 +173,8 @@ public String getSelectedAccount() {
     }
 
     public void showAccountPickerDialog() {
+        Account selectedAccount = getAccountPickerCurrentAccount();
+        intentChooseAccount.putExtra("selectedAccount",selectedAccount);
```
Comment (lognaturel) [inline collect_app/src/main/java/org/odk/collect/android/utilities/gdrive/GoogleAccountsManager.java]: Please add a space after the comma.

File: collect_app/src/main/java/org/odk/collect/android/utilities/gdrive/GoogleAccountsManager.java [hunk 2]
```diff
@@ -188,6 +192,19 @@ public void selectAccount(String accountName) {
...
+        String selectedAccountName = getSelectedAccount();
+        if (selectedAccountName == null || selectedAccountName.isEmpty()) {
+            Account[] googleAccounts = credential.getAllAccounts();
+            if (googleAccounts != null && googleAccounts.length > 0) {
+                selectedAccountName = googleAccounts[0].name;
+            } else {
+                return null;
+            }
+        }
+        return new Account(selectedAccountName,"com.google");
```
Comment (lognaturel) [inline collect_app/src/main/java/org/odk/collect/android/utilities/gdrive/GoogleAccountsManager.java]: Please add a space after the comma.

**Review by lognaturel — COMMENTED** [review]
Thanks very much @marcparicio and @srv-twry for suggesting the nice improvement!

Very small style comments inline and otherwise this is ready to merge.

**Comment by ghost** [comment]
Perfect, I updated the branch with the changes!!

**Comment by mmarciniak90** [comment]
Verified cases:
* Dialog on `Server Settings`
* Dialog on `Get Blank Form`
* Dialog on `Send Finalized Form`

@marcparicio  I noticed that toast `Account selection cancelled` is not visible with this changes.
@lognaturel This is a bug but I realized that this toast was not consistent on all views where `Choose an account` dialog is visible. 
Toast appears only on `Server Settings` view. In other views, the dialog is not visible at all.
I can create a separate issue for this toast or issue need to be updated if we want to keep it in the old form.

**Review by lognaturel — APPROVED** [review]

**Comment by lognaturel** [comment]
Thank you @marcparicio!

@mmarciniak90 Do you think that toast is actually useful? I'd tend to say we should remove it everywhere.

**Comment by mmarciniak90** [comment]
@lognaturel I agree, toast is not necessary.

**Tested with success!**
Verified on Android: 4.1, 4.2, 4.4, 5.1, 6.0, 7.0, 8.0

@opendatakit-bot unlabel "needs testing"
@opendatakit-bot label "behavior verified"

**Pull request merged**